<BlockSwitcher>

<Block name="Async/Await">

```swift
func confirmSignUp(for username: String, with confirmationCode: String) async {
    do {
        try await Amplify.Auth.confirmSignUp(
            for: username,
            confirmationCode: confirmationCode
        )
        print("Confirm signUp succeeded")
    } catch let error as AuthError {
        print("An error occurred while confirming sign up \(error)")
    } catch {
        print("Unexpected error: \(error)")
    }
}
```

</Block>

<Block name="Combine">

```swift
func confirmSignUp(for username: String, with confirmationCode: String) -> AnyCancellable {
    Amplify.Publisher.create {
        try await Amplify.Auth.confirmSignUp(
            for: username,
            confirmationCode: confirmationCode
        )
    }.sink {
        if case let .failure(authError) = $0 {
            print("An error occurred while confirming sign up \(authError)")
        }
    }
    receiveValue: { _ in
        print("Confirm signUp succeeded")
    }
}
```

</Block>

</BlockSwitcher>
